### DISCUSSION TITLE
I need to make a change to react-data-table-component package

### DISCUSSION THREAD

**Discussion by alexk400** [community/community#49830] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

`function mergeObject(target, source, options) {
    var destination = {};
	if (options.isMergeableObject(target)) {
		getKeys(target).forEach(function(key) {
          	destination[key] = cloneUnlessOtherwiseSpecified(target[key], options);
		});
	}
	getKeys(source).forEach(function(key) {
		if (propertyIsUnsafe(target, key)) {
			return
		}

		if (propertyIsOnObject(target, key) && options.isMergeableObject(source[key])) {
			destination[key] = getMergeFunction(key, options)(target[key], source[key], options);
		} else {
			destination[key] = cloneUnlessOtherwiseSpecified(source[key], options);
		}

        var newdest=destination[key];
        if(typeof newdest === 'function') 
        {
             destination[key]=newdest();
        }
        

	});
	return destination
}`

This 3 line code allow to make all style properties can be anonymous function for further customization.
        var newdest=destination[key];
        if(typeof newdest === 'function') 
             destination[key]=newdest();
        
     This allow me to customie color with further calculation 
      color: (opacity=1) =>  `rgba(255, 255, 255, ${opacity})`
      
      Atpresent the code cant decipher anonymous function.
      
      I dont know how to make change in github in react-data-table-component package
      
      Thanks
      alexk

**Comment by GurteX** [comment]
i wonder too

**Comment by alexk400** [comment]
hi

it all works in my local visual code. But every time  i use in different
projects i want  it in main master.
i only try to copy chart example and found style property do not accept
anonymous function. All it needed is  merge method.
But it was in a .dev.js file in my node_modules. And change work.

But cant find source code file for it. I dont see mergeobject function in
src directory


regards
alexk
surrey

**Comment by alexk400** [comment]
hi

its mergeobject in deepmerge.
it needs a filter callback before it merges. so it can check if its a
function then run function and then merge.
callback here to check if its a function. Tomorrow it could be some other
calculation

thanks
alexk
surrey

**Comment by alexk400** [comment]
hi

its mergeobject in deepmerge.
it needs a filter callback before it merges. so it can check if its a
function then run function and then merge.
callback here to check if its a function. Tomorrow it could be some other
calculation

thanks
alexk
surrey

**Comment by alexk400** [comment]
hi

I have another question .
I spent whole day hacking the package. But i have not  figured out. i am so
noob.

This is the issue. I want someway to close that expanded row without
scrollin back up as my detail page will be long.

I broke my brain to figure this out but no avail.

let me know if there is a quckway. i have close button in detail panel. if
i click i should be able to close the current row.

But cant figure out how to make expanded flag in row component inside table

let me know

regards
alexk
surrey

**Comment by alexk400** [comment]
hi

I have another question .
I spent whole day hacking the package. But i have not  figured out. i am so
noob.

This is the issue. I want someway to close that expanded row without
scrollin back up as my detail page will be long.

I broke my brain to figure this out but no avail.

let me know if there is a quckway. i have close button in detail panel. if
i click i should be able to close the current row.

But cant figure out how to make expanded flag in row component inside table

let me know

regards
alexk
surrey

**Comment by alexk400** [comment]
hi

I need to learn that part of forking and updating.

I did fork on deepmerge but i can create only branch on master.

And then try to build react-data-table-component by pointing my branch url

but never able to add deep merge code into
node_modules\react-data-table-component\dist\react-data-table-component.dev.js
to test.
its big learning curve.

but solved my close detail panel.

I had to pass parent table data useState variable to expandedComponent and
update the expanded variable and when it load again delete that expanded

There should be example so people dont bang their head as Github no issue
tab or featurerequest tab and no way to communicate to author

[image: image.png]

Atleast author should have discord server for the package.

Next issue i see is when there is no data table disappeared.  MY plan is
use table as placeholder and add buttons inside.

Thats all. I need to figure out the issue by providing fake data if  there
is server issue or no data. So i get table showing zero data.

regards
alex
surrey

**Comment by alexk400** [comment]
Also its a bad design. Row need to have two keys. Row.id Row.data

**Comment by alexk400** [comment]
Hi

That package owner dont have feature request or issues on his github.
I am using his package instead of material-react-datatable

I need few things.  Expand icon ( checkbox) -- has to be externalised.

Basically i want to implement my own icon in that with specific action on
table.

for example when it mobile or tablet mode i will have expand but in desktop
mod i will have edit icon

Also top left corner all expand or Add Icon. Also it can be scrollable
button for mobile as you hover over. so user can do many things.

Leave the customization to users.  provide a way to customize layout.

I can fork and do all that but i want the package owner add a prop for it.
Just like for expand he has prop and i have done crud form with that.

let me know

regards
alexk
surrey
